Packs/Office365MessageTrace/Integrations/Office365MessageTraceEventCollector/Office365MessageTraceEventCollector.yml

```diff
@@ -0,0 +1,138 @@
+category: Analytics & SIEM
+sectionorder:
+- Connect
+- Collect
+commonfields:
+  id: Office 365 Message Trace Event Collector
+  version: -1
+configuration:
+- additionalinfo: The URL for the reporting web service.
+  defaultvalue: https://reports.office365.com/ecp/reportingwebservice/reporting.svc
+  display: URL
+  name: url
+  required: false
+  type: 0
+  section: Connect
+- additionalinfo: Azure AD tenant ID
+  display: Tenant ID
+  name: tenant_id
+  required: true
+  type: 0
+  section: Connect
+- additionalinfo: Azure AD application client ID
+  display: Client ID
+  name: client_id
+  required: true
+  type: 0
+  section: Connect
+- additionalinfo: Azure AD application client secret
+  display: ""
+  displaypassword: Client Secret
+  name: client_secret
+  required: true
+  hiddenusername: true
+  type: 9
+  section: Connect
+- additionalinfo: X.509 certificate thumbprint for certificate-based authentication (optional)
+  display: Certificate Thumbprint
+  name: certificate_thumbprint
+  required: false
+  type: 0
+  section: Connect
+- additionalinfo: Private key for certificate-based authentication (optional)
+  display: Private Key
+  name: private_key
+  required: false
+  type: 14
+  section: Connect
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+  section: Connect
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+  section: Connect
+- section: Collect
+  display: First fetch time
+  name: first_fetch
+  defaultvalue: 1 day
+  type: 0
+  required: true
+- section: Collect
+  additionalinfo: Office 365 can take up to 24 hours to process message trace events. Specify how many minutes to delay event processing (1440 minutes = 24 hours). Higher values ensure events are fully processed but increase latency.
+  defaultvalue: "1440"
+  display: Processing Delay (minutes)
+  name: processing_delay
+  required: false
+  type: 0
+description: Office 365 Message Trace integration allows you to retrieve email message tracking information from the Office 365 reporting web service. You can search for message traces by date range, sender, recipient, message ID, and status to troubleshoot email delivery issues.
+display: Office 365 Message Trace Event Collector
+name: Office 365 Message Trace Event Collector
+script:
+  script: ''
+  commands:
+  - arguments:
+    - description: Start date for the search in ISO format (YYYY-MM-DDTHH:MM:SSZ). If not provided, defaults to 48 hours ago.
+      name: start_date
+    - description: End date for the search in ISO format (YYYY-MM-DDTHH:MM:SSZ). If not provided, defaults to current time.
+      name: end_date
+    - description: Date range for the search (e.g., "24 hours", "7 days"). Only used if start_date and end_date are not provided.
```

Comment:
Add A validation to ensure only one date specification method is used at a time.